gensim/models/phrases.py

```diff
@@ -227,12 +243,17 @@ def export_phrases(self, sentences, out_delimiter=b' ', as_tuples=False):
             then you can debug the threshold with generated tsv
         """
         for sentence in sentences:
-            s = [utils.any2utf8(w) for w in sentence]
             last_bigram = False
             vocab = self.vocab
             threshold = self.threshold
             delimiter = self.delimiter  # delimiter used for lookup
             min_count = self.min_count
+
+            if self.recode_to_utf8:
+                s = [utils.any2utf8(w) for w in sentence]
+            else:
+                s = [utils.any2utf8(w) for w in sentence] if self.is_input_bytes else list(sentence)
```

Comment:
As @piskvorky mentioned, suppose we had `recode_to_utf8=False`, and learn vocab was called with latin2 bytestrings, and latin2 bytestrings are sent to `__getitem__`. We'd silently recode the latin2 bytestrings to utf8 in `__getitem__` and lookup would fail, even though it should succeed.